src/main/java/net/fabricmc/loom/util/ZipUtils.java

```diff
@@ -251,4 +249,16 @@ public static boolean transform(Path zip, Map<String, UnsafeUnaryOperator<byte[]
 	public interface UnsafeFunction<T, R> {
 		R apply(T arg) throws IOException;
 	}
+
+	private static <T> Map<String, UnsafeUnaryOperator<T>> collectTransformersStream(Stream<Pair<String, UnsafeUnaryOperator<T>>> transforms) {
+		Map<String, UnsafeUnaryOperator<T>> map = new HashMap<>();
+		Iterator<Pair<String, UnsafeUnaryOperator<T>>> iterator = transforms.iterator();
+
+		while (iterator.hasNext()) {
+			Pair<String, UnsafeUnaryOperator<T>> next = iterator.next();
+			map.put(next.getLeft(), next.getRight());
```

Comment:
This has an inconsistent duplicate handling mechanism from the previous version, where this uses the latest of duplicates while the previous uses the earliest of duplicates. Should we just add a warning if there is any duplicate?